source/contracts/Controlled.sol

```diff
@@ -8,6 +8,9 @@ import 'libraries/token/ERC20Basic.sol';
 
 contract Controlled is IControlled {
     IController internal controller;
+    // Inidcates whether ETH may be extracted from this contract by the controller. It should only be allowed when ETH cannot legitimately be held by the contract
+    bool internal mayExtractETH = true;
+    mapping(address => bool) tokenExtractionDisallowed;
```

Comment:
Hmm, same argument as above, but much harder to implement with the mapping.  :/  Can we make it so that if the derived class doesn't take any action all token withdraws are disallowed, and if they take action then this blacklist is used?

Reply:
EtherDelta combines the ETH and Tokens into a single mapping where ETH is address `0x0000...`.  Consider doing the same here rather than having two variables that are used in very similar ways.  This would make ETH "less special".

Reply:
Doing this would mean that ETH needs to be blacklisted rather then whitelisted if I'm understanding correctly. Otherwise I do like it.

Reply:
W/ regard to the first question that's easy enough to do, although I think it will just result in boilerplate where every subclass of controlled turns on Token withdraws through extraction won't it?

Reply:
Yes (to both).

Another option is to be a bit less generalized and assert that all tokens can be extracted from all contracts by default _except_ `ReputationToken`, `Cash` and `ETH`.  Then this would become a whitelist for tokens in that set that _can_ be withdrawn.

Note: This solution will start to crumble a bit once we add support for markets denominated in other tokens.  I don't believe any of our contracts are designed to hold anything beyond those 3 right now though, so maybe it is "good enough"?

🤔 more after typing the above, Reputation is a bit tricky since there isn't just one reputation token, but there _is_ one reputation token per universe.  I wonder if we could make this somehow "do the right thing" and for contracts that shouldn't allow REP withdraw limit that constraint to only REP for the current universe?  Maybe an abstract method on this class that derivatives need to implement for `getReputationToken()`?

Or perhaps an abstract method for `getProtectedTokens() returns (address[])`?  Then every contract would be forced to make the cognitive choice as to what tokens are protected.  Many would return `[]`, but that is OK because at least we had to be explicit about it and thing about it for each contract (compiler won't let us "forget" one).

Reply:
I like the `getProtectedTokens() returns (address[])` option. This is actually how I first wrote this. I also like that the subclass protection is isolated in a special function then too instead of jammed into initialize.

Reply:
Just noting since I'm pivoting now, the abstract method route ended up being a bust since `Delegator` is a subclass of `Controlled` and there didn't seem to be a clean way to make this work because of that. In particular the nature of `Delegator` is such that since it must provide a `address _target = controller.lookup(controllerLookupName);` method that method will always be called for any delegated contracts. Adding delegation to that method to get the real contract's protected tokens value started to err into no longer being clean.

Reply:
Actually, thinking more on this I do like this pattern still though. I think I'll continue this route but unfortunately it won't have compiler enforcement

Reply:
Ugh, again nm the method at the base would still be inherited. We'd have to make a new class for everything to inherit from. Backing out to previous types of implementations using storage with the considerations we've talked about here